File: server/ScrumDone.Api/Data/Assignment.cs
using ScrumDone.Api.Data.Common;

namespace ScrumDone.Api.Data
{
    public class Assignment : IHasCreatedAt, IHasUpdatedAt, IHasSoftDelete
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public DateTimeOffset? DueDate { get; set; }
        public decimal? TimeEstimate { get; set; }

        public DateTimeOffset CreatedAt { get; set; }
        public DateTimeOffset UpdatedAt { get; set; }
        public bool IsDeleted { get; set; }
        public DateTimeOffset? DeletedAt { get; set; }

        public Guid ProjectId { get; set; } 
        public Guid? SprintId { get; set; }
        public Guid StatusId { get; set; }
        public Guid PriorityId { get; set; }
        public Guid? ParentAssignmentId { get; set; }
        public List<File> Attachments { get; set; } = new();
        public Project Project { get; set; }
        public Sprint? Sprint { get; set; }
        public AssignmentStatus Status { get; set; }
        public AssignmentPriority Priority { get; set; }
        public List<AssignmentUserMTMRelation> Assignees { get; set; } = new();
        public List<AssignmentAssignmentLabelMTMRelation> Labels { get; set; } = new();
        public List<Message> Comments { get; set; } = new();
        public Assignment? ParentAssignment { get; set; }
        public List<Assignment> SubAssignments { get; set; } = new();
    }
}

File: client/src/types/assignment.ts
import type { UserSummary } from './user';

// Bazowe typy domenowe
export type AssignmentStatus = {
  id: string;
  name: string;
  hexColor: string;
};

export type AssignmentPriority = {
  id: string;
  name: string;
  hexColor: string;
};

export type AssignmentLabel = {
  id: string;
  name: string;
  hexColor: string;
};

// Modele odpowiedzi API
export type Assignment = {
  id: string;
  name: string;
  description: string;
  createdAt: string;
  updatedAt: string;
  dueDate: string | null;
  timeEstimate: string;
  status: AssignmentStatus;
  priority: AssignmentPriority | null;
  assignees: UserSummary[];
  labels: AssignmentLabel[];
  subtaskIds: string[];
  projectId: string;
  projectName: string;
  sprintId: string | null;
  parentAssignmentId: string | null;
};

// DTO do tworzenia i edycji
export type CreateAssignmentDto = {
  name: string;
  description?: string;
  dueDate?: string | null;
  timeEstimate?: string;
  statusId: string;
  priorityId: string;
  assigneeIds?: string[];
  labelIds?: string[];
  projectId: string;
  sprintId?: string | null;
  parentAssignmentId?: string | null;
};

export type UpdateAssignmentDto = Partial<CreateAssignmentDto>;

// Specjalistyczne DTO dla PUT
export type AssignmentAssigneesUpdateDto = {
  assigneeIds: string[];
};

export type AssignmentLabelsUpdateDto = {
  labelIds: string[];
};

// Zapytania
export type AssignmentQueryParams = {
  SprintIds?: string[];
  ProjectIds?: string[];
  Backlog?: boolean;
  AssigneeIds?: string[];
  PriorityIds?: string[];
  StatusIds?: string[];
  LabelIds?: string[];
  DueFrom?: string; // ISO DateTime
  DueTo?: string;   // ISO DateTime
  ExcludeNoDeadline?: boolean;
  Page?: number | string;
  Limit?: number | string;
};

// Odpowiedź paginowana
export type PaginatedAssignmentsResponse = {
  items: Assignment[];
  page: string;
  pageSize: string;
  totalCount: string;
  totalPages: string;
  hasNextPage: boolean;
  hasPreviousPage: boolean;
};